switchapi/es2k/switch_pd_fdb.c

```diff
@@ -550,66 +553,74 @@ switch_status_t switch_pd_l2_rx_forward_table_entry(
+          LNW_L2_FWD_RX_TABLE_ACTION_RX_L2_FWD_LAG_AND_RECIRCULATE, tdi_status);
       goto dealloc_resources;
     }
 
-    status = tdi_table_action_data_allocate(table_hdl, action_id, &data_hdl);
-    if (status != TDI_SUCCESS) {
+    tdi_status =
+        tdi_table_action_data_allocate(table_hdl, action_id, &data_hdl);
+    if (tdi_status != TDI_SUCCESS) {
       krnlmon_log_error(
           "Unable to get action allocator for ID: %d, "
           "error: %d",
-          action_id, status);
+          action_id, tdi_status);
       goto dealloc_resources;
     }
 
-    status = tdi_data_field_id_with_action_get(
+    tdi_status = tdi_data_field_id_with_action_get(
         table_info_hdl, LNW_ACTION_RX_L2_FWD_LAG_PARAM_LAG_ID, action_id,
         &data_field_id);
-    if (status != TDI_SUCCESS) {
+    if (tdi_status != TDI_SUCCESS) {
       krnlmon_log_error("Unable to get data field id param for: %s, error: %d",
-                        LNW_ACTION_RX_L2_FWD_LAG_PARAM_LAG_ID, status);
+                        LNW_ACTION_RX_L2_FWD_LAG_PARAM_LAG_ID, tdi_status);
       goto dealloc_resources;
     }
 
-    status = tdi_data_field_set_value(
+    tdi_status = tdi_data_field_set_value(
         data_hdl, data_field_id,
         (api_l2_rx_info->rif_handle &
          ~(SWITCH_HANDLE_TYPE_LAG << SWITCH_HANDLE_TYPE_SHIFT)));
-    if (status != TDI_SUCCESS) {
+    if (tdi_status != TDI_SUCCESS) {
       krnlmon_log_error("Unable to set action value for ID: %d, error: %d",
-                        data_field_id, status);
+                        data_field_id, tdi_status);
       goto dealloc_resources;
     }
 
-    status = tdi_table_entry_add(table_hdl, session, target_hdl, flags_hdl,
-                                 key_hdl, data_hdl);
-    if (status != TDI_SUCCESS) {
+    tdi_status = tdi_table_entry_add(table_hdl, session, target_hdl, flags_hdl,
+                                     key_hdl, data_hdl);
+    if (tdi_status != TDI_SUCCESS) {
       krnlmon_log_error("Unable to add %s table entry, error: %d",
-                        LNW_L2_FWD_RX_TABLE, status);
+                        LNW_L2_FWD_RX_TABLE, tdi_status);
       goto dealloc_resources;
     }
   } else {
     /* Delete an entry from target */
-    status =
+    tdi_status =
         tdi_table_entry_del(table_hdl, session, target_hdl, flags_hdl, key_hdl);
-    if (status != TDI_SUCCESS) {
+    if (tdi_status != TDI_SUCCESS) {
       krnlmon_log_error("Unable to delete %s entry, error: %d",
-                        LNW_L2_FWD_RX_TABLE, status);
+                        LNW_L2_FWD_RX_TABLE, tdi_status);
       goto dealloc_resources;
     }
   }
 
+  tdi_status_t dealloc_status;
 dealloc_resources:
-  status = tdi_switch_pd_deallocate_resources(flags_hdl, target_hdl, key_hdl,
-                                              data_hdl, session, entry_add);
-  return switch_pd_tdi_status_to_status(status);
+  dealloc_status = tdi_switch_pd_deallocate_resources(
```

Comment:
Re your first part of the comment -- we have error codes with value zero? zero generally means success case

Reply:
In the first two cases, we jumped to `dealloc_resources` because the function we just called returned non-zero status (an error code). _Error code_ is shorter than _non-zero status code_, and the word _status_ occurs in three other places in the same sentence. I chose _error code_ in the interests of readability.